scripts/nayduck.py

```diff
@@ -67,13 +70,28 @@ def _parse_args():
     return args
 
 
-def get_curent_sha():
-    return subprocess.check_output(['git', 'rev-parse', 'HEAD'], text=True)
+def get_sha(branch: str):
+    try:
+        sha = subprocess.check_output(['git', 'rev-parse', branch], text=True)
+
+    except subprocess.CalledProcessError as e:
+        remote_branch = 'remotes/origin/' + branch
+        print(
+            f"Couldn't find a local branch \'{branch}\'. Trying remote: {remote_branch}"
+        )
+        sha = subprocess.check_output(['git', 'rev-parse', remote_branch],
+                                      text=True)
 
+    return sha
 
-def get_current_branch():
-    return subprocess.check_output(['git', 'rev-parse', '--abbrev-ref', 'HEAD'],
-                                   text=True)
+
+def get_branch(sha: str = ""):
+    if sha:
+        return subprocess.check_output(['git', 'name-rev', sha],
+                                       text=True).strip().split(' ')[-1]
```

Comment:
Out of curiosity why do you pick the last one? It may be worth printing some warning message if there is more than one too.